wasm/jit/engine.go

```diff
@@ -434,40 +400,39 @@ func (e *engine) exec(f *compiledWasmFunction) {
 			// Meaning that the current frame exits
 			// so restore the caller's frame.
 			e.callFramePop()
-		case jitCallStatusCodeCallWasmFunction:
-			// This never panics as we made sure that the index exists for all the referenced functions
-			// in a module.
-			nextFunc := e.compiledWasmFunctions[e.functionCallIndex]
-			// Calculate the continuation address so we can resume this caller function frame.
-			currentFrame.continuationAddress = currentFrame.wasmFunction.codeInitialAddress + e.continuationAddressOffset
-			currentFrame.continuationStackPointer = e.stackPointer + nextFunc.resultCount - nextFunc.paramCount
-			// Create the callee frame.
-			frame := &callFrame{
-				continuationAddress: nextFunc.codeInitialAddress,
-				wasmFunction:        nextFunc,
-				// Set the base pointer to the beginning of the function params
-				stackBasePointer: e.stackBasePointer + e.stackPointer - nextFunc.paramCount,
+		case jitCallStatusCodeCallFunction:
```

Comment:
I consolidate host function calls with normal wasm function calls. This reduced the cost of checking isHost in the assembly, plus the cost of doing fully native function calls between wasm functions I will do later.

Reply:
change this to a code comment or put in RATIONALE.md as it is a great design decision and worth storing.